addons/uom/models/uom_uom.py

```diff
@@ -226,37 +71,36 @@ def _compute_quantity(self, qty, to_unit, round=True, rounding_method='UP', rais
             return qty
         self.ensure_one()
 
-        if self != to_unit and self.category_id.id != to_unit.category_id.id:
-            if raise_if_failure:
-                raise UserError(_(
-                    'The unit of measure %(unit)s defined on the order line doesn\'t belong to the same category as the unit of measure %(product_unit)s defined on the product. Please correct the unit of measure defined on the order line or on the product. They should belong to the same category.',
-                    unit=self.name, product_unit=to_unit.name))
-            else:
-                return qty
-
```

Comment:
I think this should be somewhat kept in some form. The issue is that if you have two uoms that have nothing to do with each other with non-`1` factor, it will lead to *silent* broken conversions.

Reply:
It may indeed lead to some meaningless calculations. But the idea is that this flexibility is _intended_ as I understood when I discussed the specs with the POs. In some business cases, such conversions make sense, when some goods are bought from vendors in some unit but then sold to customers in a totally different unit. Also, the new relative unit field is not required. It's just an optional field to help users organize their units it they want, so I think we can't totally rely on it as an alternative to UoM categories.